pydantic/fields.py

```diff
@@ -335,65 +337,105 @@ def Field(
     :param max_length: only applies to strings, requires the field to have a maximum length. The
-      schema will have a ``maxLength`` validation keyword
-    :param frozen: a boolean which defaults to True. When False, the field raises a TypeError if the field is
-      assigned on an instance.  The BaseModel Config must set validate_assignment to True
+      schema will have a `maxLength` validation keyword
+    :param frozen: a boolean which defaults to False. When True, the field raises a TypeError if the field is
+      assigned on an instance. The BaseModel Config must set validate_assignment to True
     :param pattern: only applies to strings, requires the field match against a regular expression
-      pattern string. The schema will have a ``pattern`` validation keyword
+      pattern string. The schema will have a `pattern` validation keyword
     :param discriminator: only useful with a (discriminated a.k.a. tagged) `Union` of sub models with a common field.
       The `discriminator` is the name of this common field to shorten validation and improve generated schema
     :param repr: show this field in the representation
     :param json_schema_extra: extra dict to be merged with the JSON Schema for this field
     :param strict: enable or disable strict parsing mode
     :param validate_default: whether the default value should be validated for this field
+    :param const: removed, use `Literal` instead'
+    :param unique_items: removed, use `Set` instead
+    :param allow_mutation: deprecated, use `frozen` instead
+    :param regex: removed, use `Pattern` instead
+    :param **extra: removed
     """
+    # Check deprecated & removed params of V1.
+    # This has to be removed deprecation period over.
+    if const:
+        raise PydanticUserError('`const` is removed. use `Literal` instead', code='field-removed-param')
+    if min_items:
+        warn('`min_items` is deprecated and will be removed. use `min_length` instead', DeprecationWarning)
+        if min_length is None:
+            min_length = min_items
+    if max_items:
+        warn('`max_items` is deprecated and will be removed. use `max_length` instead', DeprecationWarning)
+        if max_length is None:
+            max_length = max_items
+    if unique_items:
+        raise PydanticUserError(
+            (
+                '`unique_items` is removed, use `Set` instead'
+                '(this feature is discussed in https://github.com/pydantic/pydantic-core/issues/296)'
+            ),
+            code='field-removed-param',
+        )
+    if allow_mutation is False:
+        warn('`allow_mutation` is deprecated and will be removed. use `frozen` instead', DeprecationWarning)
+        frozen = True
+    if regex:
+        raise PydanticUserError('`regex` is removed. use `Pattern` instead', code='field-removed-param')
+    if extra:
+        raise PydanticUserError('Extra keyword arguments are not allowd on `Field`', code='field-removed-param')
```

Comment:
* rename to `deprecated_kwargs`
* this has been replaced by `json_schema_extra`
* should therefore be a warning not an error.